### ISSUE TITLE
`Lint/MixedCaseRange` raises an error for some non-literal ranges

### ISSUE THREAD

**Issue by jaynetics** [rubocop/rubocop#12259]
The begin or end of Regexp ranges can be non-literals (e.g. subsets, unicode properties, posix classes, codepoint lists).

It is better not to do that because it sometimes leads to weird Regexp behavior. There should perhaps be a cop against it. Either way, `MixedCaseRange` should not raise errors when encountering such constructs.

It will probably suffice to change [this comparison](https://github.com/rubocop/rubocop/blob/986d047b8a6d4163329cdd4925726eae7885b6e6/lib/rubocop/cop/lint/mixed_case_range.rb#L98) to `!= :literal`.

## Expected behavior

No error.

## Actual behavior

```
undefined method `begin' for nil
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:105:in `rewrite_regexp_range'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:51:in `block (2 levels) in on_regexp'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/base.rb:377:in `correct'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/base.rb:181:in `add_offense'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:50:in `block in on_regexp'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:65:in `block (2 levels) in each_unsafe_regexp_range'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:60:in `reject'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:60:in `block in each_unsafe_regexp_range'
[...]/gems/3.3.0+0/gems/regexp_parser-2.8.1/lib/regexp_parser/expression/methods/traverse.rb:75:in `block in each_expression_without_index'
[...]/gems/3.3.0+0/gems/regexp_parser-2.8.1/lib/regexp_parser/expression/subexpression.rb:28:in `each'
[...]/gems/3.3.0+0/gems/regexp_parser-2.8.1/lib/regexp_parser/expression/subexpression.rb:28:in `each'
[...]/gems/3.3.0+0/gems/regexp_parser-2.8.1/lib/regexp_parser/expression/methods/traverse.rb:74:in `each_expression_without_index'
[...]/gems/3.3.0+0/gems/regexp_parser-2.8.1/lib/regexp_parser/expression/methods/traverse.rb:13:in `each_expression'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:57:in `each_unsafe_regexp_range'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/lint/mixed_case_range.rb:49:in `on_regexp'
[...]/gems/3.3.0+0/gems/rubocop-1.56.4/lib/rubocop/cop/commissioner.rb:107:in `public_send'
```

## Steps to reproduce the problem

```sh
echo '/[a-[bc]]/' > foo.rb
rubocop --only Lint/MixedCaseRange -d foo.rb
```

## RuboCop version

```
1.56.4 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 3.3.0) [arm64-darwin22]
```
